### DISCUSSION TITLE
How to get typehints for this kind of equality test?

### DISCUSSION THREAD

**Discussion by santi-mir** [microsoft/pyright#9600] [Q&A] [answered]
I might be doing something wrong, but trying this code:

```python

def a()->Literal['a','b','c']:
    return 'a'

if a()=='x':
    print('x')
```

it does not show an error, not does it with anything else replaced on `'x'` like `{ }` probably just uses string for the typehint. 
[Playground link here.](https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAySMApiAIYA2AUNQCYnBTkAUAlALQB8RpFlAbQDk5IQBohAI3FCAxkIC6ALmpQ1UECRgBXECigihtJE1ZsAvOYDeAXxXqoCEKhgshADyFsgA)

I expect that we could get typehint both for different literals, and for different data types.

**Comment by erictraut** [comment, marked as answer]
If you would like pyright to detect condition tests where the operands appear to have no type overlap, you can enable the `reportUnnecessaryComparison` check. This check is not enabled by default.

**Reply by santi-mir** [reply]
thanks, yeah at least in some cases it would be the only way to catch a typo.
